clients/web/src/lib/connection/transport.tsx

```diff
@@ -121,12 +121,9 @@ export function addStreamListneners(
 
     const spansUpdate = update.spansUpdate;
     if (spansUpdate && spansUpdate.spanEvents.length > 0) {
-      setMonitorData(
-        "spans",
-        produce((clonedSpans) =>
-          updatedSpans(clonedSpans, spansUpdate.spanEvents)
-        )
-      );
+      setMonitorData("spans", (spans) => [
+        ...updatedSpans(spans, spansUpdate.spanEvents),
```

Comment:
this is mutating the entire array. Every time this `setMonitorData` is called, we replace the entire array. How about wrapping it in a `reconcile()` so only the needed keys are mutated? 🤔

Reply:
I will try to use Solid's `reconcile` here

Reply:
For the use of reconcile the `updatedSpans` function and the receiving piece on the calls tab might need updates. I would like to postpone this change to the calls virtualization PR.